### PR TITLE
Detailed tx summaries

### PR DESCRIPTION
**Pull request by Oscar-Pepper**
for debugging and advanced zingo-cli users
currently, detailed tx summary only adds nullifiers but there may be other things that will be useful to add in the future such as the notes spent in a tx etc.

### CODE REVIEW

**Review by fluidvanadium — DISMISSED** [review]
ok!

**Comment by fluidvanadium** [comment]
i will test it out in zingocli when i get a chance

File: zingolib/src/wallet/data.rs [hunk 1]
```diff
@@ -783,7 +788,96 @@ pub mod summaries {
         }
     }
 
-    /// TODO: doc comment
+    /// Basic transaction summary interface
+    pub trait TransactionSummaryInterface {
```
Comment (zancas) [inline zingolib/src/wallet/data.rs:792]: Why is this logic packaged in a separate trait?   Why aren't these methods inherent to a `TransactionSummary`?

Comment (zancas) [reply zingolib/src/wallet/data.rs:792]: I am fine with it as is, just curious.

**Review by zancas — DISMISSED** [review]

**Comment by zancas** [comment]
@Oscar-Pepper it seems that there's a CI fail.

**Pull request merged**